app/helpers/loss_report_helper.rb

```diff
@@ -0,0 +1,211 @@
+    # following 'if' structure is intended to increase readability by eliminating nests
+    return text_output.concat("#{@counter}.".ljust(4)+"#{change}: #{element}".ljust(60) + path + "\n") if hide_items == false && disp == 'text'
+    return text_output.concat("#{@counter}.".ljust(4)+"#{change}: ".ljust(3) + path + "\n") if hide_items == true && disp == 'text'
+    return json_output["#{@counter}. #{change}: #{path}"] = element if disp == 'json'
+  end
+
+  def hash_map(hash)
+    buckets = Array.new
+    hash.each do |key,val|
+      if val.is_a? Hash then hash_map(val).each do |item|
+        item['path'] = key + '/' + item['path']
+        buckets << item end
+      else
+        buckets << {'path'=> key, 'value'=> val}
+      end
+    end
+    buckets
+  end
+
+  def prepare_collections(concept_id, umm_c_version)
+    original_collection_native_xml = cmr_client.get_concept(concept_id,token, {})
+    return false if !original_collection_native_xml.success?
+
+    content_type = original_collection_native_xml.headers.fetch('content-type').split(';')[0]
+    original_collection_native_hash = Hash.from_xml(original_collection_native_xml.body)
+    translated_collection_umm_json = cmr_client.translate_collection(original_collection_native_xml.body, content_type, "application/vnd.nasa.cmr.umm+json;version=#{umm_c_version}", skip_validation=true)
+    return false if !translated_collection_umm_json.success?
+
+    translated_collection_native_xml = cmr_client.translate_collection(JSON.pretty_generate(translated_collection_umm_json.body), "application/vnd.nasa.cmr.umm+json;version=#{umm_c_version}", content_type,  skip_validation=true)
+    return false if !translated_collection_native_xml.success?
+
+    translated_collection_native_hash = Hash.from_xml(translated_collection_native_xml.body)
+    return original_collection_native_xml.body, translated_collection_native_xml.body, original_collection_native_hash, translated_collection_native_hash, content_type
+  end
+
+
+  def hash_navigation(path, hash)
+    # Passed a path string and the hash being navigated. This method parses the path string and
+    # returns the array/value at the end of the path
+    current_path = String.new
+    path.split('/').each do |key|
+      if hash.is_a?(Array)
+        return hash, current_path
+      elsif hash.key?(key) && hash.is_a?(Hash)
+        current_path += "/#{key}"
+        hash = hash[key]
+      elsif !hash.key?(key) && key != ''
+        return path_exists = false, "#{current_path}/#{key}"
+      end
+    end
+    return hash, current_path
+  end
+
+  def array_comparison(path, original_hash, converted_hash)
+    pre_translation_array = hash_navigation(path, original_hash)[0]
+    post_translation_array = hash_navigation(path, converted_hash)[0]
+
+    pre_translation_array == false ? pre_translation_array = Array.new : pre_translation_array = Array.wrap(pre_translation_array)
```

Comment:
I should probably make a comment to better explain this in the code since the purpose of these conditionals has evolved since the 2311 PR. The current purpose is now to notice a `false` output from `hash_navigation`. This is because `hash_navigation` will return `false` if the path passed to `hash_navigation` doesn't exist. A path may not exist if, for instance, an array that was in the original collection was completely removed during translation and therefore doesn't exist in the translated collection. So in order for the code following the conditionals in question to function properly, array-wrapping a `false` will cause a `false` line item to be output to the loss-report. Therefore, the `pre_translation_array` needs to be redefined as an empty array.

Reply:
Ah, then the last bit can be struck.